graphql_compiler/query_pagination/query_parameterizer.py

```diff
@@ -1,61 +1,288 @@
+        new_parameters: The parameters to use with the new_ast
+    """
+    if not isinstance(query_ast, (FieldNode, InlineFragmentNode, OperationDefinitionNode)):
+        raise AssertionError(
+            f'Input AST is of type "{type(query_ast).__name__}", which should not be a selection.'
+        )
+
+    if len(query_path) == 0:
+        return _add_pagination_filter_at_node(
+            query_ast, pagination_field, directive_to_add, extended_parameters
+        )
+
+    if query_ast.selection_set is None:
+        raise AssertionError(f"Invalid query path {query_path} {query_ast}.")
+
+    found_field = False
+    new_selections = []
+    for selection_ast in query_ast.selection_set.selections:
+        new_selection_ast = selection_ast
+        field_name = get_ast_field_name(selection_ast)
+        if field_name == query_path[0]:
+            found_field = True
+            new_selection_ast, new_parameters = _add_pagination_filter(
+                selection_ast,
+                query_path[1:],
+                pagination_field,
+                directive_to_add,
+                extended_parameters,
+            )
+        new_selections.append(new_selection_ast)
+
+    if not found_field:
+        raise AssertionError(f"Invalid query path {query_path} {query_ast}.")
+
+    new_ast = copy(query_ast)
+    new_ast.selection_set = SelectionSetNode(selections=new_selections)
+    return new_ast, new_parameters
+
+
+def _make_binary_filter_directive_node(op_name: str, param_name: str) -> DirectiveNode:
+    """Make a binary filter directive node with the given binary op_name and parameter name."""
+    return DirectiveNode(
+        name=NameNode(value="filter"),
+        arguments=[
+            ArgumentNode(name=NameNode(value="op_name"), value=StringValueNode(value=op_name),),
+            ArgumentNode(
+                name=NameNode(value="value"),
+                value=ListValueNode(values=[StringValueNode(value="$" + param_name)]),
+            ),
+        ],
+    )
+
+
+def generate_parameterized_queries(
+    schema_info: QueryPlanningSchemaInfo,
+    query: ASTWithParameters,
+    vertex_partition: VertexPartitionPlan,
+    parameter_value: Any,
+) -> Tuple[ASTWithParameters, ASTWithParameters]:
+    """Generate two parameterized queries that can be used to paginate over a given query.
 
     Args:
         schema_info: QueryPlanningSchemaInfo
-        query_ast: Document, query that is being paginated.
-        parameters: dict, list of parameters for the given query.
+        query: the query to parameterize
+        vertex_partition: pagination plan
+        parameter_value: the value of the parameter used for pagination
 
     Returns:
-        ParameterizedPaginationQueries namedtuple
+        next_page: Ast and params for next page.
+        remainder_ast: Ast and params for remainder.
     """
-    raise NotImplementedError()
+    query_type = get_only_query_definition(query.query_ast, GraphQLError)
```

Comment:
Is `query_type` an appropriate name here? I don't think this is a type based on how it's used, it looks more like the root AST definition node for the query.